packages/ckeditor5-widget/tests/widgettypearound/widgettypearound.js

```diff
@@ -250,82 +280,842 @@ describe( 'WidgetTypeAround', () => {
 		} );
 	} );
 
-	describe( 'detection and CSS classes of widgets needing the typing around support', () => {
-		it( 'should detect widgets that are a first child of the parent container', () => {
-			setModelData( editor.model, '<blockWidget></blockWidget><paragraph>foo</paragraph>' );
+	describe( 'typing around view widgets using keyboard', () => {
+		let model, modelSelection, eventInfoStub, domEventDataStub;
 
-			const viewWidget = viewRoot.getChild( 0 );
+		beforeEach( () => {
+			model = editor.model;
+			modelSelection = model.document.selection;
+		} );
+
+		describe( '"fake caret" activation', () => {
```

Comment:
There isn't any tests for up/down arrows.

Reply:
Also no test for <kbd>Shift</kbd> + <kbd>Arrow</kbd>

Reply:
Also second step is missing (from fake caret to fully selected widget), as it is tested in "deactivation".

Reply:
> Also second step is missing (from fake caret to fully selected widget), as it is tested in "deactivation".

[These are tested by these 2 tests.](https://github.com/ckeditor/ckeditor5/blob/i/6693/packages/ckeditor5-widget/tests/widgettypearound/widgettypearound.js#L567-L615)